src/FormParserFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace React\Http;
+
+use React\Http\Parser;
+
+class FormParserFactory
+{
+    /**
+     * @param Request $request
+     * @return Parser\ParserInterface
+     */
+    public static function create(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        if (
+            !isset($headers['content-type']) ||
+            (!isset($headers['content-type']) && !isset($headers['content-length']))
```

Comment:
Are you sure that's what it's testing? :)

How about `!isset($headers['content-type'], $headers['content-length'])` ?